compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Transforms.h

```diff
@@ -29,6 +29,15 @@ LogicalResult createLogicalObjectFifoLink(
 LogicalResult hoistForAffineApplyOp(RewriterBase &rewriter,
                                     affine::AffineApplyOp applyOp);
 
+/// Normalize the loop bounds of the `scf.for` operation to lowerbound == 0 and
+/// step == 1.
+LogicalResult normalizeLoopBounds(RewriterBase &rewriter, scf::ForOp forOp);
+
+/// Normalize the loop bounds of the `scf.forall` operation to lowerbound == 0
+/// and step == 1.
+LogicalResult normalizeLoopBounds(RewriterBase &rewriter,
+                                  scf::ForallOp forallOp);
```

Comment:
Instead maybe have just one API `normalizeLoopBounds(RewriterBase&, LoopLikeInterface)`. Within the API maybe multiplex to the respective static functions ?

Reply:
I addressed this in the upstream PR, we can discuss further there.